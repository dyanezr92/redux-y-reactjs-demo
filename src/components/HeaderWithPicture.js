import React from 'react'
import PropTypes from 'prop-types'
import TitleCounter from './TitleCounter'

const HeaderWithPicture = ({ title, src, numComments, numComplaints }) => (
  <div>
    <h1 className="text-center">{ title }</h1>
    <TitleCounter numElements={ numComments } elementsName="comments" />
    <TitleCounter numElements={ numComplaints } elementsName="complaints" />

    <img alt="Header" src={ src } className="img-thumbnail img-responsive center-block"/>
  </div>
);

HeaderWithPicture.propTypes = {
  title: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired,
  numComments: PropTypes.number.isRequired,
  numComplaints: PropTypes.number.isRequired
}

export default HeaderWithPicture
