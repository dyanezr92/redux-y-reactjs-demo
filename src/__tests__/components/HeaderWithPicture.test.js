import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components

import HeaderWithPicture from '../../components/HeaderWithPicture';
import finn from '../../static/finn.gif'

describe('<HeaderWithPicture title="Awesome Finn" src={ finn } />', () => {
  describe('render()', () => {
    test('renders the header', () => {
      const wrapper = shallow(<HeaderWithPicture title="Awesome Finn" src={ finn } numComments={1} numComplaints={2} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
