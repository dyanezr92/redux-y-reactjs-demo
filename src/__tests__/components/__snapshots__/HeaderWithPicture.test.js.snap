// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderWithPicture title="Awesome Finn" src={ finn } /> render() renders the header 1`] = `
<div>
  <h1
    className="text-center"
  >
    Awesome Finn
  </h1>
  <TitleCounter
    elementsName="comments"
    numElements={1}
  />
  <TitleCounter
    elementsName="complaints"
    numElements={2}
  />
  <img
    alt="Header"
    className="img-thumbnail img-responsive center-block"
    src="finn.gif"
  />
</div>
`;
